modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2722,16 +2722,19 @@ private Expression<?>[] _getSelectExpressions(
 						relatedDynamicObjectDefinitionTable)
 				).from(
 					relatedDynamicObjectDefinitionTable
-				).innerJoinON(
-					relatedExtensionDynamicObjectDefinitionTable,
-					relatedExtensionDynamicObjectDefinitionTable.
-						getPrimaryKeyColumn(
-						).eq(
-							relatedDynamicObjectDefinitionTable.
-								getPrimaryKeyColumn()
-						)
 				);
 
+				if (!objectRelationship.isSelf()) {
+					joinStep = joinStep.innerJoinON(
+						relatedExtensionDynamicObjectDefinitionTable,
+						relatedExtensionDynamicObjectDefinitionTable.
+							getPrimaryKeyColumn(
+							).eq(
+								relatedDynamicObjectDefinitionTable.
+									getPrimaryKeyColumn()
+							));
+				}
```

Comment:
@pedro-oliveira446 could you please provide here an example of this query for self relationship?

Thank you!

Reply:
Old query :
select O_20096_Ticket.c_ticketId_, 
	   O_20096_Ticket.title_, 
       (
			select count(O_20096_Ticket.c_ticketId_) 
            from O_20096_Ticket 
				inner join O_20096_Ticket_x on O_20096_Ticket_x.c_ticketId_ = O_20096_Ticket.c_ticketId_ 
                inner join ObjectEntry on ObjectEntry.objectEntryId = O_20096_Ticket.c_ticketId_ 
                inner join R_20096Ticket_Ticket_rel on R_20096Ticket_Ticket_rel.c_ticketId_2 = O_20096_Ticket.c_ticketId_ 
			where R_20096Ticket_Ticket_rel.c_ticketId_1 = O_20096_Ticket_x.c_ticketId_
		) as agreg 
from O_20096_Ticket 
	inner join O_20096_Ticket_x on O_20096_Ticket.c_ticketId_ = O_20096_Ticket_x.c_ticketId_ 
where O_20096_Ticket.c_ticketId_ = 38753;


New query :
`select O_20096_Ticket.c_ticketId_, 
	   O_20096_Ticket.title_, 
       (
			select count(O_20096_Ticket.c_ticketId_) 
            from O_20096_Ticket 
				inner join ObjectEntry on ObjectEntry.objectEntryId = O_20096_Ticket.c_ticketId_ 
                inner join R_20096Ticket_Ticket_rel on R_20096Ticket_Ticket_rel.c_ticketId_2 = O_20096_Ticket.c_ticketId_ 
			where R_20096Ticket_Ticket_rel.c_ticketId_1 = O_20096_Ticket_x.c_ticketId_
		) as agreg 
from O_20096_Ticket 
	inner join O_20096_Ticket_x on O_20096_Ticket.c_ticketId_ = O_20096_Ticket_x.c_ticketId_ 
where O_20096_Ticket.c_ticketId_ = 38753`